Upcoming Newsletter

jdavis opened the pull request:
Here lies the upcoming newsletter. We can start collecting news, issues, pull requests, etc.

- [x] Terminal UI with termkey/unibilium
- [x] Finished removing `long_u` by @elmart
- [x] Build system updates (builds on FreeBSD?) by @jszakmeister 
- [x] New tab events
- [x] Removal of easy mode
- [x] Removal of edcompatible and ttyfast
- [x] Simpler handling of "always-on"/"always-off" options: [2002](https://github.com/neovim/neovim/pull/2002)
- [x] Garbage collection speedup: [1761](https://github.com/neovim/neovim/pull/1761)
- [x] EndOfBuffer highlight group: [1926](https://github.com/neovim/neovim/pull/1926)
- [x] Merge of vimdiff with nvim -d: [1849](https://github.com/neovim/neovim/pull/1849)
- [x] Meta keys section
- [x] Windows build
- [x] Terminal emulation: [2076](https://github.com/neovim/neovim/pull/2076)
- [x] True colors: [2198](https://github.com/neovim/neovim/pull/2198)
- [x] ICCF
- [x] Current GUIs
- [x] Job control API changes
- [x] Blackduck
- [x] Neovim related projects
- [x] @Pyrohh's  PR [2008](https://github.com/neovim/neovim/pull/2008)
- [x] Vim Expect library: [2314](https://github.com/neovim/neovim/pull/2314)
- [x] Add section on `nocp` being default

fwalch commented:
Has the [list of Neovim-related projects](https://github.com/neovim/neovim/wiki/Related-projects) already been mentioned in a newsletter? It's not new, but might still be interesting. Plus, drawing people's attention to it might result in some new entries.

jdavis commented:
It hasn't; I agree in that it'd be worth sharing. I'll add a section on it!

ReneFroger commented:
Just wondering if there is any update for the newsletter of February? No need to hurry, by the way. Your effort is really appreciated. :+1:

justinmk commented:
worth a mention: nvim has intuitively made progress towards architectural recommendations made in a 2003 formal paper: https://github.com/neovim/neovim/issues/1859#issue-54996218

jdavis commented:
Thanks for the interest and words, @ReneFroger. My work has been consuming my mind quite a bit this month. I'll be returning to the newsletter soon =]

agsdot commented:
@jdavis, great news!  Look forward to reading the next newsletter.

justinmk commented:
More notes (will add to this):

- `nocp` is now the default, and `set cp` is a no-op. Also, `nvim -u NONE` works like `vim -u NONE -N`. 
    - `-N` is also ignored, and `-c` is an error.

Pyrohh commented:
- [#1820](https://github.com/neovim/neovim/pull/1820) was merged
- @elmart finished removing `long_u`, and fixing many coverity/clang analysis issues
- There have been a few PRs by @jszakmeister so that nvim now builds on FreeBSD
- Added `EndOfBuffer` highlight group [#1926](https://github.com/neovim/neovim/pull/1926).
- Added `{TabNew,TabNewEntered,TabClosed}` events [#1717](https://github.com/neovim/neovim/pull/1717)
- Removed easy mode [#1656](https://github.com/neovim/neovim/pull/1656), and merged diff mode [#1849](https://github.com/neovim/neovim/pull/1849)
- Removed `edcompatible` and `ttyfast` (the former should have similar behavior to `set cp` after [#2002](https://github.com/neovim/neovim/pull/2002) is merged)
- [#1761](https://github.com/neovim/neovim/pull/1761) was apparently [pretty popular](https://www.reddit.com/r/vim/comments/2uso8f/a_patch_fixing_vims_garbage_collection_bug_has/)
- Tons of patches by @fwalch (too many to link)

I might have missed a few notable ones, but those seemed like the most obvious.

jdavis commented:
@Pyrohh Awesome. That'll help a lot. Thanks!

justinmk commented:
Updated notes:

- `nocp` is now the default, and `set cp` is an error. Also, `nvim -u NONE` works like `vim -u NONE -N`. 
    - `-N` is also ignored, and `-c` is an error.
- Meta keys are mappable in the terminal. See https://github.com/neovim/neovim/wiki/Differences-from-Vim for details.
- Customer testimonial: https://twitter.com/telemachus/status/568918223745372160

Pyrohh commented:
> #1761 was apparently pretty popular

Probably shouldn't include the reddit discussion, since it's for Vim.

> Customer testimonial: https://twitter.com/telemachus/status/568918223745372160

User testimonial sounds better :)

tarruda commented:
> @equalsraf has a Windows build running on appveyor

The windows builld was an amazing contribution by @equalsraf :smile:

jszakmeister commented:
> The windows builld was an amazing contribution by @equalsraf

A **huge** :+1: to @equalsraf for that!

tarruda commented:
@jdavis  I'm currently working on something that some might consider a "killer feature" giving neovim a significant advantage over vim, and I expect it to be ready by friday. Just letting you know in case you were planning to release this week, as this feature would make a good addiiton to the newsletter

jdavis commented:
@tarruda Cool! I don't think I'll be able to get to it/finish it by then so we'll be able to include it!

cweagans commented:
@tarruda oooo, that sounds exciting. Have a link to a PR or a branch or something? :)

tarruda commented:
> @tarruda oooo, that sounds exciting. Have a link to a PR or a branch or something? :)

Not yet as its still too unstable. Maybe tomorrow

tarruda commented:
@cweagans neovim/neovim#2076

jdavis commented:
@tarruda You were right about it being big/interesting: https://news.ycombinator.com/item?id=9132352

tarruda commented:
@jdavis a lot of people always wanted to run terminals in vim buffers :smile:

elmart commented:
Maybe it's worth mentioning we were chosen as one of the 2014 Open Source Rookies of the year.
http://ht.ly/I1qbj

tarruda commented:
RGB colors are now supported in terminals: https://github.com/neovim/neovim/pull/2198

justinmk commented:
ICCF Holland donation: https://twitter.com/unixstickers/status/580318671341559808

jdavis commented:
@justinmk Cool! Also it looks like either you or @tarruda saw the Tweetdeck email awhile back which is great.

jdavis commented:
Just to clarify: is the meta key bindings in the terminal because Neovim now uses libtermkey?

tarruda commented:
> Just to clarify: is the meta key bindings in the terminal because Neovim now uses libtermkey?

Yes, we now use libtermkey to parse terminal keys and mouse events.

tarruda commented:
Just merged #2076

jdavis commented:
@tarruda Awesome. That'll be easier for people to try.

@elmart I added a section on Blackduck; take a look at it.

@justinmk I added the ICCF Holland section, let me know what you think.

@fmoralesc I added a section on your easy mode removal. Let me know if any details/wording needs to be changed.

Pyrohh commented:
@jdavis I think you may be running on an older version of nvim, as `:h evim-keys` has no result for me.

jdavis commented:
@Pyrohh Maybe the wording isn't clear but this:

> in Vim's documentation using `:h evim-keys`

was meant to mean that they can still read about easy mode if they run that in Vim (not nvim). I'll clarify that though.

Pyrohh commented:
Ohhhh, I missed the `N`.

jdavis commented:
@elmart @justinmk Should we also mirror the [call to arms PR](https://github.com/neovim/neovim/issues/2122) to get more awareness that reviewers are very important and that Neovim needs more?

elmart commented:
@jdavis Not sure about that. Probably, that was more of a wish than a plausible thing. I'm about to finish a PR introducing some simple tooling to make acknowledging reviewers easier. But it's clear that good reviewers must grow out of regular contributors, so I'm not sure about publishing that.

fwalch commented:
https://github.com/rogual/neovim-dot-app could be mentioned in the newsletter. I can't test it (no OSX), but from what I've read it should be quite usable.

tarruda commented:
There's a major change in the job control API: #2247, this is worth mentioning because it breaks existing plugins that use it.

jdavis commented:
@elmart Okay, that sounds good.

@fwalch I've been meaning to test it out so I'll do so then add a section on it.

@tarruda Okay, thanks for the heads up.

jdavis commented:
@tarruda I tried to break that PR down a bit and added a section on it. Let me know if I'm accurate.

@fwalch @rogual I added a section on the new GUI. Let me know what you think.

fwalch commented:
@jdavis After the discussion starting in https://github.com/neovim/neovim.github.io/issues/88#issuecomment-86940819, I noticed that nearly all of the GUI projects have been added after the last newsletter came out. So maybe we could do a section on just "GUI efforts" to not single out just one GUI? Chronologically, I think it was like this:

* @tarruda's experimental Python/GTK GUI
* (new TUI)
* Atom plugin: vim-mode (exited before, but didn't work with `master`. Has not been mentioned in a previous newsletter, I think)
* Atom-shell based: neovim.as
* Rust IDE: solidoak
* OS X: neovim.app

The other new OS X GUI project, nvox, is experimental, and its readme now points to neovim.app.

jdavis commented:
@fwalch I think that's reasonable and makes sense. I think the TUI should be mentioned on its own so maybe a TUI and a GUI section?

fwalch commented:
@jdavis :+1:

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,250 @@
+
+### Related Projects
+
+The [Neovim Wiki][related-main-wiki] contains a page titled [Related
```
Perhaps this could link to the main wiki page, and...

jdavis replied:
What do you mean? The `[related-main-wiki]` is linked with [this line](https://github.com/neovim/neovim.github.io/pull/82/files#diff-e6d5661bcb3c99d1e3afa66e3aa0e12fR280)

Pyrohh replied:
This is what happens when I start reading at the bottom.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,250 @@
+the projects might still be in progress.
+
+Do you know of another project that can be listed? Don't be afraid to add a link
+yourself!
```
This could link to the project page (it's really begging for a hyper link IMO).

jdavis replied:
In addition to the link that I have on [line 32](https://github.com/neovim/neovim.github.io/pull/82/files#diff-e6d5661bcb3c99d1e3afa66e3aa0e12fR32)?

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,250 @@
+@tarruda has finished [his pull request][term-pr] adding a new `:terminal`
+command which when ran launches a full terminal emulator. The terminal is using
+Neovim's native buffers and windows which means the user can enter [normal
+mode][term-normal] by pressing `<C-\><C-n>`.
```
Now that it's been documented, perhaps you could append something like the following to this paragraph:

```
See [`:h nvim-terminal-emulator`][help-terminal-emulator] for more information.

...

[help-terminal-emulator]: http://neovim.org/doc/user/nvim_terminal_emulator.html#nvim-terminal-emulator
```

jdavis replied:
Good idea now that the docs are out!

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,250 @@
+Similar to `evim`/`vim -y` described above, [`vimdiff`][vimdiff-docs] is the
+same as `vim -d` which starts Vim in diff mode.
+
+@Pyrohh created a [pull request][vimdiff-pr] that merged these two different
```
I don't want to sound like I'm tooting my own horn, but conceptually the vimdiff PR was only half of the picture. The other half was https://github.com/neovim/neovim/pull/2008 (vimdiff just had proportionally more documentation).

jdavis replied:
No, that's a fair point. I didn't know of that PR and must have glossed over it. I'll relook at that PR tomorrow.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,250 @@
+
+### Blackduck 2014 Open Source Rookie
+
+[Blackduck][blackduck-site], a company that helps other corporations in the
```
It looks like this is two words. From https://www.blackducksoftware.com/about:

```
At **Black Duck**, we believe the future ...
```

jdavis replied:
You are right. Fixing it.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,250 @@
+bought increases the amount donated to [ICCF Holland][iccf-site], Bram
+Moolenaar's charity.
+
+Since they've been added, $106 has been raised which Unixstickers [just
```
Many countries use dollars, so perhaps it should be USD.

jdavis replied:
I believe I am? Unixstickers site defaults to USD (top right) so I just assumed that the amount on the side of each [sticker page](http://www.unixstickers.com/stickers/software_stickers/neovim-text-editor-logo-shaped-sticker) was USD?

Pyrohh replied:
Oh no, I mean a literal `USD` in place of the `$`

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,283 @@
+### Major Change to Improve Job API
+
+@tarruda is continuing to refine the new job API. His work has continued in
+[this latest pull request][job-pr] which hasn't been merged yet. This might be a
```
This has now been merged.

jdavis replied:
I didn't comment, but I updated this btw.

oni-link commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,285 @@
+interface using [libtermkey][tui-libtermkey] and [unibilium][tui-unibilium].
+
+Libtermkey is a library by @leonerd that "allows easy processing of keyboard
+entry from terminal-based programs." It has the ability to handle special keys,
```
Switch positions of characters in `."`?

jdavis replied:
I had it the other way but decided against it. I'll update it.

tarruda commented:
I just merged neovim/neovim#2314, a small vimscript library that aims to give [expect](http://en.wikipedia.org/wiki/Expect)-like capabilities to Neovim(builds on top of job control, of course). Unfortunately I don't have time to document it properly yet, but I've added a [plugin](https://github.com/neovim/neovim/blob/master/contrib/neovim_gdb/neovim_gdb.vim) that integrates Neovim with gdb as an example(which I'm already using for debugging Neovim :smile: )

jdavis commented:
@tarruda Awesome. I'll look into that.

@justinmk @fwalch @Pyrohh Besides the rest of what needs to be done in the top comment, are there any other big missing pieces that haven't been mentioned?

fwalch commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,285 @@
+
+### ICCF Holland Donations
+
+Mentioned in the last letter, [Unixstickers][iccf-unixstickers] has added
```
Maybe "As mentioned ... newsletter"?

jdavis replied:
Updated

fwalch commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,285 @@
+entry from terminal-based programs." It has the ability to handle special keys,
+[UTF-8 combining][tui-combining] and more.
+
+Unibilium is a library that interfaces with [terminfo][tui-terminfo]. Terminfo
```
Since you mention leonerd for libtermkey, you could mention mauke here as well.

jdavis replied:
Good point. Added a mention.

fwalch commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,285 @@
+[tui-pr]: https://github.com/neovim/neovim/pull/1820
+[tui-libtermkey]: http://www.leonerd.org.uk/code/libtermkey/
+[tui-unibilium]: https://github.com/mauke/unibilium
+[tui-bindings]: https://github.com/neovim/neovim/wiki/Differences-from-Vim#features
```
Link to http://neovim.org/doc/user/vim_diff.html#nvim-features-new instead?

jdavis replied:
Updated

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,285 @@
+
+#### Additional Keybindings
+
+As a result of using libtermkey, Neovim can now bind [meta keys][tui-meta] to
```
I think this makes it sounds like it's primarily meta keys, when it's conceptually more than just that IMO.

jdavis replied:
I added the details on Ctrl-Shift.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,285 @@
+Neovim continues to merge in the upstream Vim patches that are submitted to the
+Vim mailing list.
+
+Since the last newsletter, [60 pull requests][patch-search] worth of patches
```
Since this was written, I think a few more have been added, so perhaps `more than 60 pull requests ...`?

jdavis replied:
Updated.

Pyrohh commented:
@jdavis 

> @justinmk @fwalch @Pyrohh Besides the rest of what needs to be done in the top comment, are there any other big missing pieces that haven't been mentioned?

I think the expect library would be good to include, although that's up to you of course.

justinmk commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,322 @@
+
+[Black Duck][blackduck-site], a company that helps other corporations in the
+adoption of open source software, chose Neovim as one of the projects to feature
+of this last year. The company annually recognizes some of the top new open
```
"to feature of this last year" is a bit weird. Is that intentional wording?

jdavis replied:
I did not intend to word it like that. I wrote it one way then changed it and accidentally a word.

justinmk commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,322 @@
+
+## Development News
+
+### Replacing Terminal UI
```
can you change this to:

> ### Terminal UI (TUI)

To introduce people to the term "TUI" which is very useful for us.

jdavis replied:
That's a good way to introduce it. Added.

justinmk commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,322 @@
+### Replacing Terminal UI
+
+@tarruda has merged his [pull request][tui-pr] to reimplement the terminal user
+interface using [libtermkey][tui-libtermkey] and [unibilium][tui-unibilium].
```
Add a sentence:

> Although the TUI implementation is included as part of Neovim, it is architecturally decoupled from the editor core: it uses the same API as external clients and GUIs.

jdavis replied:
I did not know that. Updated.

justinmk commented:
I still think we should have a pull-quote of [this](https://twitter.com/telemachus/status/568918223745372160) somewhere in the newsletter.


Also important:

- `nocp` is now the default, and `set cp` is an error. Also, `nvim -u NONE` works like `vim -u NONE -N`. 
    - `-N` is ignored, and `-C` is an error.
- @equalsraf has a Windows build running on [appveyor](https://ci.appveyor.com/project/equalsraf/neovim)
    - this was not mentioned in the previous newsletter, it's big news, though it seems like old news to us :)

Pyrohh commented:
> I still think we should have a pull-quote of this quote somewhere in the newsletter (in the margin or at the top?).

In a newsletter full of interesting, tangible happenings in Neovim's development, I don't think an anecdote adds anything of value. I agree that the development environment is very friendly and relaxed, but I don't think one person's experience is substantial enough for a newsletter.

jdavis commented:
@Pyrohh I still plan on adding that expect section. It's the last item on the list.
@justinmk Yup, the Windows build is still pending. I'll add a bullet for nocp.

jdavis commented:
@Pyrohh @justinmk I think Justin means something similar to the last newsletter linked [here](https://github.com/neovim/neovim.github.io/blame/master/_posts/2014-11-07-november-newsletter.md#L8). It doesn't need a section or explanation, but probably an attribute this time.

justinmk commented:
Anecdotes are undoubtedly useful for connecting to the reader. yes, it is emotional and non scientific, and that is the purpose: marketing. Newsletters for a project are like press releases. Part of it is showing off, and gaining mindshare, besides the informational part. It's all part of dealing with humans :)

Pyrohh commented:
> Anecdotes are undoubtedly useful for connecting to the reader. yes, it is emotional and non scientific, and that is the purpose: marketing.

Yes of course, but do you not think that just the information is enough?

> Newsletters for a project are like press releases. Part of it is showing off, and gaining mindshare, besides the informational part. It's all part of dealing with humans :)

Well we can show off just by stating the new and exciting features available, as is already being done. Anyways, it's just one quote, so I don't want to go on a crusade over it. I don't think I'd do very well in marketing...

jdavis commented:
An interesting example of what @justinmk is talking about is that someone (either Justin or @tarruda) has been retweeting a lot more on the [Neovim Twitter](https://twitter.com/neovim) account. Before they started monitoring it, I tried to keep the retweets to a minimum; only doing it when there hadn't been an interesting tweet in a few weeks. But since these more recent retweets, there has been like way more talk and chatter about Neovim. It seems like a lot more people are trying it than before. Granted there a lot more tangible benefits now like the TUI, true colors and `:terminal` but it's the same idea.

jdavis commented:
@equalsraf I want to make sure I get the Windows stuff straight. So right now it is possible to compile it? I can see that there is [this Gist](https://gist.github.com/equalsraf/4685f7aef022a36c26d5) which has some info. Then of course there are the two issues: [#810](https://github.com/neovim/neovim/pull/810) and [1749](https://github.com/neovim/neovim/issues/1749). What's the detailed status of it?

jdavis commented:
@Pyrohh I could have left out the list and just linked to the docs, but listing them all will at least ensure that there won't be an excuse for that single person that uses `rgview` to be livid because we didn't warn them :smile:

fwalch commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,343 @@
+
+Welcome to the fifth newsletter for Neovim.
+
+## General News
```
### Gitter chatroom for Neovim

In addition to the mailinglist and the IRC channel, Neovim now has a [chatroom on Gitter.im][gitter-room]. Besides using the interface, you can also [use your IRC client](irc.gitter.im) to connect to it.

-- ... or something like that :grin:

jdavis replied:
Added!

justinmk commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,343 @@
+
+#### Removal of "Easy Mode"
+
+@fmoralesc also opened a [pull request][easy-pr] to remove `evim`. Which is as
```
sentence wording

jdavis replied:
Fixed

justinmk commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,343 @@
+@Pyrohh opened [an issue][in-discussion] to discuss the removal of such
+aliases. All of the aliases had corresponding flags that could be used instead.
+
+After some discussion and polls it was decided to  remove the lesser used ones.
```
technically we didn't have a poll. but there was thorough discussion.

jdavis replied:
Right, not a formal poll, but I was referring to @fmoralesc's [query](https://github.com/neovim/neovim/issues/1646#issuecomment-66815330)

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+Now in addition to the [mailing list][gitter-list] and IRC channel, you can hop
+into the [Neovim chat room][gitter-link] provided through
+[Gitter.im][gitter-link]. It has a convenient web interface but also allows you
+to connect using your favorite IRC client.
```
Maybe this?
```
... connect using your [favorite IRC client](https://irc.gitter.im/).

jdavis replied:
Updated!

Pyrohh replied:
Huh, that's weird. I got a notification that you pushed the change, and it says you did on your profile, but it's not showing up...

jdavis replied:
High five! We caused a bug on GitHub.

Pyrohh replied:
The newsletter was too long :grin:

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+
+Although the TUI implementation is included as part of Neovim, it is
+architecturally decoupled from the editor core: it uses the same API as external
+clients and GUIs.
```
Aren't clients implicitly "external", at least in this case? If it's not clear, then I suppose keeping "external" is fine.

jdavis replied:
@justinmk Might have to chime in on that paragraph.

justinmk replied:
the TUI in this case is an internal client to some degree. Lua VimL may also be.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+mode][term-normal] by pressing `<C-\><C-n>`.
+
+This is meant to replace and provide more capabilities than the already existing
+Vim command, `:shell`, which just launches a shell. More explanation on the
```
Could link to the `:shell` documentation for vim:

```
[`:shell`][vim-doc-shell]

...

[vim-doc-shell]: http://vimdoc.sourceforge.net/htmldoc/various.html#:shell
```

jdavis replied:
Added that link

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+Neovim's native buffers and windows which means the user can enter [normal
+mode][term-normal] by pressing `<C-\><C-n>`.
+
+This is meant to replace and provide more capabilities than the already existing
```
Given that `:shell` doesn't seem to exist in Nvim, perhaps it should be something like:
```
This replaces `:shell`, but is meant to provide more capabilities ...
```
or (on the line below):
```
... than the already existing Vim command , `shell` (which has been removed), ...
```

I like the first one more, but these are just rough ideas.

jdavis replied:
Yeah, it could be more explicit. Also, I like the second one.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+@tarruda.
+
+The complete documentation for this feature can be found under [`:h
+nvim-terminal-emulator`][term-docs].
```
I just realized we should probably not be using abbreviations for help items, so I think `:help ...` would be better.

jdavis replied:
I've used `:h` before and there doesn't seem to be any confusion but we can move to that to be more explicit.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+### Adding a EndOfBuffer Highlight Group
+
+@mhinz wanted a way to [remove ~ after the file ends][highlight-idea]. He
+proposed a new [:highlight][highlight-docs] group, called
```
I think `:highlight` should be surrounded by `

jdavis replied:
Fixed

Pyrohh commented:
> @Pyrohh I could have left out the list and just linked to the docs, but listing them all will at least ensure that there won't be an excuse for that single person that uses rgview to be livid because we didn't warn them 
> 
> Plus I found it absolutely fascinating that that many aliases existed just to open vim; I only knew of the major ones.

:+1: It's a bit ridiculous. By the way, I think you can still emulate the old functionality using `v:progpath` or `v:progname`.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+A more detailed explanation of what true colors means as well as terminals that
+support it can be found in [this Gist][color-info] that was written by @XVilka.
+
+To enable true colors, set the environment variable
```
I think it should be `export the env... `.

Pyrohh replied:
Also it should be probably be `enable true color support`, as the user's terminal doesn't necessarily support true color, and the user's colorscheme isn't necessarily configured for it.

jdavis replied:
Is the distinction important? You can set an environment variable using `VAR=blah command` or `export VAR=blah`. Exporting (at least to me) implies it is available for other commands that are ran after. While you could use "setting" for the first one but not for the latter while "export" only makes sense for the second. Really, I don't think it matters.

Your second fix is good though so people don't just assume it'll work regardless of terminal.

Pyrohh replied:
An exported environment variable is necessarily set, but an environment variable is not necessarily exported, so I think you're right.

One thing though: the way it's worded might make people think that the following is enough:
```
export NVIM_TUI_ENABLE_TRUE_COLOR
nvim
```

when in reality, the first results in nothing being exported (at least according to `env` in both `bash` and `mksh`). Given that the corresponding code just checks if the environment variable exists, then maybe the sentence being discussed could say something like:
```
... set the environment variable `NVIM_YAYA` to `1`, e.g.

export NVIM_YEYE=1
nvim

or just launch nvim like this (assuming you're using a bourne/korn shell derivative):

NVIM_YEYE=1 nvim
```

It might just be better to assume people aren't using `tcsh` or `csh`.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+### True Colors
+
+With the changes to the TUI described above, this allowed @tarruda to
+make a [pull request][color-pr] to add true colors to terminals that support it.
```
Maybe "... make a pull request adding [true color](https://en.wikipedia.org/wiki/Color_depth#True_color_.2824-bit.29) for supporting terminals." ?

jdavis replied:
Fixed

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+In a separate [pull request][vimdiff-pr], @Pyrohh, merged
+[`vimdiff`][vimdiff-docs] and `vim -d` which were two different ways of viewing
+a diff in Vim.
+
```
Could you merge this section with `Special interfaces cleanup`? This sentence:
```
... which were two different ways of viewing a diff in Vim.
```
makes it seem like `vimdiff` is different than `vim -d`, when they're the same thing (it's just that `vimdiff` and `vim -d` were used interchangeably before, the former more than the latter). Conceptually it's just another "Special interface".

jdavis replied:
Sure, since you are the author of both, I think that's fine. I'm going to keep the easy mode section since you didn't make that PR.

Pyrohh replied:
> I'm going to keep the easy mode section since you didn't make that PR.

Well there's that, but the reason I didn't comment on easy mode is because it's more than just a symlink and some docs (runtime file, some code IIRC).

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,392 @@
+- [`view`][in-view]: Read-only mode.
+- [`gvim`][in-gvim]: Using the GUI.
+- [`gex`][in-gex]: GUI in Ex mode.
+- [`gview`][in-gview]: GUI in read-only mode.
```
Regarding this line and the above line:

To match the verbiage used in the examples below, maybe it should be `GUI and in ...`.

jdavis replied:
Fixed

jdavis commented:
Okay, all of the main bullet points are done in the top comment. Someone needs to look over the Windows section. Right now this newsletter is already much longer than previous ones (which means sense since it has been awhile):

```
$ wc -l _posts/*-newsletter.md
     282 _posts/2014-06-06-june-newsletter.md
     229 _posts/2014-07-04-july-newsletter.md
     252 _posts/2014-09-06-september-newsletter.md
     316 _posts/2014-11-07-november-newsletter.md
     471 _posts/2015-04-03-april-newsletter.md
    1550 total
```

If anyone has any other suggestions for sections/items, let me know. I'll rebase this soon and we'll be able to release it Monday, probably?

justinmk commented:
LGTM

jdavis commented:
Awesome, thanks @justinmk. We even could release it tomorrow.

Pyrohh commented:
> If anyone has any other suggestions for sections/items, let me know. I'll rebase this soon and we'll be able to release it Monday, probably?

[I left a comment](https://github.com/neovim/neovim.github.io/pull/82/files#r27770752), which I assume you just didn't notice due to how many there already are here.

jdavis commented:
I think all that remains is a title. Also, this PR isn't updating even though the branch has [updated](https://github.com/neovim/neovim.github.io/compare/master...jdavis:newsletter?expand=1)?

Pyrohh commented:
I wonder if it will update if the merge conflicts are fixed (or are there no merge conflicts?).

jdavis commented:
@Pyrohh There aren't any merge conflicts now. The conflicts started showing up like 10 commits ago. I noticed it wasn't updating like 5 commits ago. I figured a rebase and push --force would shake it lose but it didn't help.

Pyrohh commented:
And just so I don't forget, I think we should release on Monday. Maybe this is irrational, but I think that more people might read this on a Monday, or basically any day which isn't Sunday.

jdavis commented:
@Pyrohh Yeah, plus it is Easter as well. I wanted to finish it this past Friday but I wasn't able to. Looks like Monday it is. Also, it needs a title. Do you know of any funny quotes that have happened in the issue tracker these past months?

jdavis commented:
Also, I got a response from GitHub. They confirmed something broke and are fixing it.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+
+Libtermkey is a library by @leonerd that "allows easy processing of keyboard
+entry from terminal-based programs." It has the ability to handle special keys,
+[UTF-8 combining][tui-combining] and more.
```
`UTF-8 key combinations`? Maybe @tarruda knows...

jdavis replied:
This was @leonerd's wording himself: http://www.leonerd.org.uk/code/libtermkey/

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+
+Unibilium is a library by @mauke that interfaces with [terminfo][tui-terminfo].
+Terminfo is a library and database that makes it easy to interface with display
+terminals without worrying about device specific details. More details can be
```
Since most terminals are in software, "device" sounds odd here IMO as it conveys a physical terminal. Maybe `implementation` instead?

jdavis replied:
It was originally for devices which is why I used that wording: http://en.wikipedia.org/wiki/Terminfo

Pyrohh replied:
:+1:

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+### Builtin Terminal Emulation
+
+@tarruda has finished [his pull request][term-pr] adding a new `:terminal`
+command which when ran launches a full terminal emulator. The terminal is using
```
maybe ` The terminal uses`?

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
perhaps the long form of `cp` should be used, to match other options.

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
same here regarding long form

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
same here regarding long form

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
IMO this title is misleading, as it implies that `nocp` only had its default changed, when in reality a decent amount of code was removed. I think the following would be more accurate:
```
#### Removal of `compatible`
```

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,476 @@
+### Special Interfaces Cleaning
+
+Vim has a number of ways to start it through its many aliases. The complete list
+can be found under the [starting section][in-starting] on the Vimdocs.
```
(at the end of the line) maybe `in Vim's documentation`?

Pyrohh commented on _posts/2015-04-03-april-newsletter.md:
Maybe `... command. Vim's documentation ...`? I know that Vimdocs is the name of the website, but it is after all just a frontend for Vim's documentation.

jdavis commented on _posts/2015-04-03-april-newsletter.md:
The cp/compatible and nocp/nocompatible are good ideas. Updated.

jdavis commented:
I added a possible title based on @justinmk's ideas. I don't see why it couldn't be introduced now as it hasn't been yet.

Pyrohh commented:
The title is perfect, at least according to me :)

justinmk commented:
Ship it! :)

jdavis commented:
Cool, I have tweet schedule to go out tomorrow morning announcing it. Unless of course people visit the page first.

Pyrohh commented:
Make sure to post it to `/r/neovim` and `/r/vim` (unless someone beats you to it). I don't want to do it now as some people might miss it otherwise.

tarruda commented:
@jdavis the newsletter looks really nice, thanks for your work. 

> Cool, I have tweet scheduled to go out tomorrow morning announcing it. Unless of course people visit the page first.

Great, I was about to do it myself but will wait for the scheduled tweet, looking forward to see the newsletter activity :smile:

Tranquility commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,478 @@
+### Gitter.im for Neovim
+
+Now in addition to the [mailing list][gitter-list] and IRC channel, you can hop
+into the [Neovim chat room][gitter-link] provided through
```
This should have been gitter-nvim

jdavis replied:
Fixed! Thanks.

fwalch commented on _posts/2015-04-03-april-newsletter.md:
```diff
@@ -0,0 +1,478 @@
+  uses Neovim for binding and manipulating text. Take a look at [this
+  video][gui-vim-mode-video] to see it.
+- [python-client][gui-python] by @tarruda: which in addition to being a Python
+  library for communicating with Neovim, it also contains an interface using
```
`which` (or `, it`) should probably be removed

jdavis replied:
Thanks @fwalch. I've updated it.

jdavis commented:
@Pyrohh @fwalch Thanks a lot for both of your eagle eyes!
